language-support/java/codegen/BUILD.bazel

```diff
@@ -163,55 +164,68 @@ test_daml_lf_target_versions = [
 ########################################################
 
 it_daml_lf_target_versions = [
-    "1.0",
-    "1.1",
-    "1.5",
-    "latest",
+    "1.6",
+    "1.7",
+    "1.8",
 ]
 
-# This file was created using a patched version of 0.12.12.
-# See https://github.com/digital-asset/daml/pull/2443 for more details
-alias(
-    name = "integration-tests-model-1.0.dar",
-    actual = ":src/it/dar/integration-tests-model-1.0.dar",
-)
-
-# This file was created using
-# DAML_SDK_VERSION=0.12.20 daml damlc package src/it/daml/Lib.daml integration-tests-model-1.1 --target=1.1 --output src/it/dar/integration-tests-model-1.1.dar
-alias(
-    name = "integration-tests-model-1.1.dar",
-    actual = ":src/it/dar/integration-tests-model-1.1.dar",
-)
-
-# This file was created using
-# DAML_SDK_VERSION=0.13.10 daml damlc package src/it/daml/Lib.daml integration-tests-model-1.5 --target=1.5 --output src/it/dar/integration-tests-model-1.5.dar
-alias(
-    name = "integration-tests-model-1.5.dar",
-    actual = ":src/it/dar/integration-tests-model-1.5.dar",
-)
-
-daml_compile(
-    name = "integration-tests-model-latest",
-    srcs = glob(["src/it/daml/**/*.daml"]),
-    target = lf_latest_version,
-)
-
 [
-    #
-    # Compile the DAR files with and without a package to allow
-    # DAML-LF version agnostic tests to be run using generated
-    # Java classes with the same qualified name, while the DAML-LF
-    # specific tests run using Java classes generated with a
-    # package name qualified by the DAML-LF version.
-    #
     [
+        genrule(
+            name = "integration-tests-model-%s" % mangle_for_java(target),
+            srcs = ["src/it/daml"],
```

Comment:
@cocreature @sofiafaro-da @aherrmann I got a warning here:
```
WARNING: language-support/java/codegen/BUILD.bazel:175:16: input 'language-support/java/codegen/src/it/daml' to //language-support/java/codegen:integration-tests-model-1_dev is a directory; dependency checking of directories is unsound
```
Any idea how I could avoid it ? How can I rewrite the line 182 if I use  `srcs = ["src/it/daml/**"]` ?

Reply:
You can use `srcs = [glob(["src/it/daml/**/*"])]`.

Reply:
If I use `[glob(["src/it/daml/**/*"])]`, `$(location src/it/daml)` does not work.

Reply:
You will need to do something unpleasant like:

```
tar c $@ | tar x -C "$TMP_DIR" --strip-components=3
```

(No idea if this will work, but I can't think of an easier way of doing it. The others might have a better clue.)

Reply:
`locations` + crimes like https://github.com/digital-asset/daml/blob/36c267f422a9f52cada177a687a22c557d6407fe/compiler/daml-extension/BUILD.bazel#L58 should work.